atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2011,6 +2020,27 @@ private void throwIfPreCommitRequirementsNotMet(@Nullable LockToken commitLocksT
         throwIfPreCommitConditionInvalid(timestamp);
     }
 
+    private void throwIfTransactionsTableSweptBeyondReadOnlyTxn() {
+        long startTimestamp = getStartTimestamp();
+        // The schema version of current transaction does not matter. If the current transaction does not hold
+        // immutableTs lock, and we were previously on schema 4 for a range of transactions, we cannot know the state
```

Comment:
tests!

Reply:
we could technically check what was on what version, but let's roll with this for now